From d96303b3334db08d3107f498c52379515cb73107 Mon Sep 17 00:00:00 2001
From: Ted Percival <ted@midg3t.net>
Date: Thu, 3 May 2007 09:19:07 +0000
Subject: [PATCH 05/13] Do not clobber HOME environment variable.

To: Debian Bug Tracking System <submit@bugs.debian.org>
Date: Tue, 17 Oct 2006 21:57:23 +1000

During the build of dietlibc the Makefile clobbers the HOME environment
variable for the duration of the build. The HOME environment variable
is used by ccontrol to find its configuration file, but when building
dietlibc it can't find its configuration and fails to execute properly.

Here is a snippet of the build output when using ccontrol:
> /usr/bin/make prefix=/usr/lib/diet \
>           MYARCH='i386' VERSION='0.30-1'
> make[1]: Entering directory `/tmp/dietlibc-0.30'
> mkdir bin-i386
> gcc -I. -isystem include -pipe -nostdinc -Os -fomit-frame-pointer
> -falign-functions=1 -falign-jumps=1 -falign-loops=1
> -mpreferred-stack-boundary=2 -Wall -W -Wchar-subscripts
> -Wmissing-prototypes -Wmissing-declarations -Wno-switch -Wno-unused
> -Wredundant-decls -c i386/start.S -o bin-i386/start.o
> ccontrol error: reading /tmp/dietlibc-0.30/.ccontrol/config: 2
> make[1]: *** [bin-i386/start.o] Error 1

I propose changing the variable name in the Makefile to "DIETHOME" in
order to fix this. A patch to do so is attached.

-Ted
---
 Makefile |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -88,7 +88,7 @@ endif
 OBJDIR=bin-$(ARCH)
 ILIBDIR=$(LIBDIR)-$(ARCH)
 
-HOME=$(shell pwd)
+DIETHOME=$(shell pwd)
 
 WHAT=	$(OBJDIR) $(OBJDIR)/start.o $(OBJDIR)/dyn_start.o $(OBJDIR)/dyn_stop.o \
 	$(OBJDIR)/dietlibc.a $(OBJDIR)/liblatin1.a \
@@ -320,7 +320,7 @@ VERSION=dietlibc-$(shell head -n 1 CHANG
 CURNAME=$(notdir $(shell pwd))
 
 $(OBJDIR)/diet: $(OBJDIR)/start.o $(OBJDIR)/dyn_start.o diet.c $(OBJDIR)/dietlibc.a $(OBJDIR)/dyn_stop.o
-	$(CCC) -isystem include $(CFLAGS) -nostdlib -o $@ $^ -DDIETHOME=\"$(HOME)\" -DVERSION=\"$(VERSION)\" -lgcc
+	$(CCC) -isystem include $(CFLAGS) -nostdlib -o $@ $^ -DDIETHOME=\"$(DIETHOME)\" -DVERSION=\"$(VERSION)\" -lgcc
 	$(STRIP) -R .comment -R .note $@
 
 $(OBJDIR)/diet-i: $(OBJDIR)/start.o $(OBJDIR)/dyn_start.o diet.c $(OBJDIR)/dietlibc.a $(OBJDIR)/dyn_stop.o
@@ -328,7 +328,7 @@ $(OBJDIR)/diet-i: $(OBJDIR)/start.o $(OB
 	$(STRIP) -R .comment -R .note $@
 
 $(PICODIR)/diet-dyn: $(PICODIR)/start.o $(PICODIR)/dyn_start.o diet.c
-	$(LD_UNSET) $(CCC) -isystem include $(CFLAGS) -fPIC -nostdlib -o $@ $^ -DDIETHOME=\"$(HOME)\" -D__DYN_LIB -DVERSION=\"$(VERSION)\" -L$(PICODIR) -lc -lgcc $(PICODIR)/dyn_stop.o -Wl,-dynamic-linker=$(HOME)/$(PICODIR)/libdl.so
+	$(LD_UNSET) $(CCC) -isystem include $(CFLAGS) -fPIC -nostdlib -o $@ $^ -DDIETHOME=\"$(DIETHOME)\" -D__DYN_LIB -DVERSION=\"$(VERSION)\" -L$(PICODIR) -lc -lgcc $(PICODIR)/dyn_stop.o -Wl,-dynamic-linker=$(DIETHOME)/$(PICODIR)/libdl.so
 	$(STRIP) -R .command -R .note $@
 
 $(PICODIR)/diet-dyn-i: $(PICODIR)/start.o $(PICODIR)/dyn_start.o diet.c
